daemon/api_notices.go

```diff
@@ -16,28 +16,34 @@ package daemon
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"math"
 	"net/http"
+	"path/filepath"
 	"strconv"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/strutil"
 )
 
 var noticeReadInterfaces = map[state.NoticeType][]string{
 	state.ChangeUpdateNotice:   {"snap-refresh-observe"},
 	state.RefreshInhibitNotice: {"snap-refresh-observe"},
+	state.SnapRunInhibitNotice: {"snap-refresh-observe"},
 }
 
 var (
 	noticesCmd = &Command{
-		Path:       "/v2/notices",
-		GET:        getNotices,
-		ReadAccess: interfaceOpenAccess{Interfaces: []string{"snap-refresh-observe"}},
+		Path:        "/v2/notices",
+		GET:         getNotices,
+		POST:        postNotices,
+		ReadAccess:  interfaceOpenAccess{Interfaces: []string{"snap-refresh-observe"}},
+		WriteAccess: openAccess{},
```

Comment:
I think perhaps a different type of access, something like `snapClientAccess{}` might be appropriate here? Again, for the time being, we don't want to allow anyone outside of snapd/the snap client to be able to create notices. Your solution of checking `/proc/<pid>/exe` works, but the endpoint being `openAccess` suggests to me that anyone should be able to create notices of any type, neither of which are true.

Something like `snapClientAccess` could implement the check you added below while maintaining that this behaves differently from the way `openAccess` works in other places, (afaik) without restrictions about the caller (beyond it not being a snap).

Reply:
That's a good point. Furthermore, in the future, we may wish to allow snaps to create notices, in which case it would become `interfaceOpenAccess`, which is definitely not compatible with a `snapClientAccess`. Using `openAccess` as you suggest is analogous to the way `interfaceOpenAccess` grants access only to particular notice types depending on the connected interface, so there is precedent as well.

So +1 from me for your current approach. Thanks!

Reply:
How is this endpoint currently protected in pebble?

Reply:
@Meulengracht  It is not protected, it uses an access checker that similar to `openAccess`.